### PR TITLE
case sensitive set dict keys

### PR DESCRIPTION
**Pull request by aogier**
This PR allow preserve key case when setting them in existing key attributes

close: #2795

### CODE REVIEW

File: evennia/commands/default/building.py [hunk 1]
```diff
@@ -126,7 +128,17 @@ def parse(self):
...
+                    _attrs = []
+                    for match in (
+                        match
+                        for part in map(str.strip, attrs.split("/"))
+                        if part and (match := _KEY_REGEX.match(part.strip()))
+                    ):
+                        attr = match.group("attr").lower()
+                        if match.group("key"):
+                            attr += match.group("key")
+                        _attrs += [attr]
```
Comment (Griatch) [inline evennia/commands/default/building.py:140]: Please use `_attrs.append(attr)` here, it's not only faster than using the add operator, it's also more Pythonic. :)

File: evennia/commands/default/building.py [hunk 2]
```diff
@@ -126,7 +128,17 @@ def parse(self):
                     aliases = [alias.strip() for alias in aliases.split(";") if alias.strip()]
                 if "/" in objdef:
                     objdef, attrs = [part.strip() for part in objdef.split("/", 1)]
-                    attrs = [part.strip().lower() for part in attrs.split("/") if part.strip()]
```
Comment (Griatch) [inline evennia/commands/default/building.py:129]: I'm not sure I follow why this regex complexity is needed ... could we not just skip the `.lower()` here? :thinking:

**Review by Griatch — CHANGES_REQUESTED** [review]
Thanks for the PR. I'm not sure I follow why the extra regexes are needed to resolve the issue though ... we want to preserve case in the dict of the command, so it seems removing `.lower()` would be enough? Could you eleaborate? 

Either way, we'd need new unit tests to showcase the case-sensitive scenario.

File: evennia/commands/default/building.py (hunk 2 above)
Comment (aogier) [reply evennia/commands/default/building.py:129]: What I did here is to preserve a design choice (lowercase attribute names)

```
create box
You create a new Object: box.
set box/CasedAttribute = "foo"
Created attribute box/casedattribute [category:None] = foo
```

fixing a lack of expected functionality when coping w/ dicts

```
set box/dict_attr = {'foo': 'bar'}
Created attribute box/dict_attr [category:None] = {'foo': 'bar'}
set box/dict_attr['FOO'] = 'new'
Modified attribute box/dict_attr [category:None] = {'foo': 'new'}
examine box/dict_attr
-------------------------------------------------------------------------------
Attribute box/dict_attr [category=None] [type: <class 'dict'>]:

{'foo': 'new'}
```

so it has been necessary to extract the `key` part and mangling it separately from the rest of attribute name. Of course we could simply remove the `lower()`, I simply did not want to take the responsibility of introducing a breaking(ish) change :smile_cat:

File: evennia/commands/default/building.py (hunk 1 above)
Comment (aogier) [reply evennia/commands/default/building.py]: done!

File: evennia/commands/default/building.py (hunk 2 above)
Comment (Griatch) [reply evennia/commands/default/building.py:129]: Ah, that's true; the rest of the attribute name does assume case-insensitivity. Fair enough, I buy that. Thanks for elaborating!
 
I would suggest adding a code comment to explain this in the code; always good to help the reader where needed. 
Also still needs unit tests for this case ;)

**Review by Griatch — APPROVED** [review]
Thanks for the updates. Looking good now.

**Pull request merged**